src/DynamicMvvm.Uno.WinUI/Extensions/DispatcherQueueExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/License.md
+// See reference: https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/Microsoft.Toolkit.Uwp/Extensions/DispatcherQueueExtensions.cs
+
+using System;
+using System.Threading.Tasks;
+using Windows.Foundation.Metadata;
+using Microsoft.UI.Dispatching;
+using System.Runtime.CompilerServices;
+
+namespace MessageDialogService.Extensions
+{
+	internal static class DispatcherQueueExtensions
+	{
+		/// <summary>
+		/// Indicates whether or not <see cref="DispatcherQueue.HasThreadAccess"/> is available.
+		/// </summary>
+		private static readonly bool IsHasThreadAccessPropertyAvailable = ApiInformation.IsMethodPresent("Windows.System.DispatcherQueue", "HasThreadAccess");
+
+		/// <summary>
+		/// Invokes a given function on the target <see cref="DispatcherQueue"/> and returns a
+		/// <see cref="Task"/> that completes when the invocation of the function is completed.
+		/// </summary>
+		/// <param name="dispatcher">The target <see cref="DispatcherQueue"/> to invoke the code on.</param>
+		/// <param name="function">The <see cref="Action"/> to invoke.</param>
+		/// <param name="priority">The priority level for the function to invoke.</param>
+		/// <returns>A <see cref="Task"/> that completes when the invocation of <paramref name="function"/> is over.</returns>
+		/// <remarks>If the current thread has access to <paramref name="dispatcher"/>, <paramref name="function"/> will be invoked directly.</remarks>
+		public static Task EnqueueAsync(this DispatcherQueue dispatcher, Action function, DispatcherQueuePriority priority = DispatcherQueuePriority.Normal)
+		{
+			// Run the function directly when we have thread access.
+			// Also reuse Task.CompletedTask in case of success,
+			// to skip an unnecessary heap allocation for every invocation.
+			if (IsHasThreadAccessPropertyAvailable && dispatcher.HasThreadAccess)
+			{
+				try
+				{
+					function();
+
+					return Task.CompletedTask;
+				}
+				catch (Exception e)
+				{
+					return Task.FromException(e);
+				}
+			}
+
+			static Task TryEnqueueAsync(DispatcherQueue dispatcher, Action function, DispatcherQueuePriority priority)
+			{
+				var taskCompletionSource = new TaskCompletionSource<object?>();
+
+				if (!dispatcher.TryEnqueue(priority, () =>
```

Comment:
I think it would be better to use a local function instead of a lambda here.

Reply:
We can probably remove the header of the file if we change the code significantly?
Hmmm not sure, might as well keep it.

Reply:
I'll keep the header just since the inspiration came from their toolkit